pkg/acsengine/engine.go

```diff
@@ -123,7 +125,31 @@ var dcos2TemplateFiles = []string{dcos2BaseFile, dcosAgentResourcesVMAS, dcosAge
 var kubernetesTemplateFiles = []string{kubernetesBaseFile, kubernetesAgentResourcesVMAS, kubernetesAgentResourcesVMSS, kubernetesAgentVars, kubernetesMasterResources, kubernetesMasterVars, kubernetesParams, kubernetesWinAgentVars, kubernetesWinAgentVarsVMSS}
 var swarmTemplateFiles = []string{swarmBaseFile, swarmParams, swarmAgentResourcesVMAS, swarmAgentVars, swarmAgentResourcesVMSS, swarmAgentResourcesClassic, swarmBaseFile, swarmMasterResources, swarmMasterVars, swarmWinAgentResourcesVMAS, swarmWinAgentResourcesVMSS}
 var swarmModeTemplateFiles = []string{swarmBaseFile, swarmParams, swarmAgentResourcesVMAS, swarmAgentVars, swarmAgentResourcesVMSS, swarmAgentResourcesClassic, swarmBaseFile, swarmMasterResources, swarmMasterVars, swarmWinAgentResourcesVMAS, swarmWinAgentResourcesVMSS}
-var openshiftTemplateFiles = append(kubernetesTemplateFiles, openshiftNodeScript, openshiftMasterScript)
+var openshiftTemplateFiles = append(
+	kubernetesTemplateFiles,
+	openshiftNodeScript,
+	openshiftMasterScript,
+	openshift39NodeScript,
+	openshift39MasterScript,
+)
+
+func getOpenshiftMasterShAsset(version string) string {
+	switch version {
+	case common.OpenShiftVersion3Dot9Dot0:
+		return openshift39MasterScript
+	default:
+		return openshiftMasterScript
```

Comment:
shouldn't the default correspond to `OpenShiftDefaultVersion`?

Reply:
Actually, thinking on this a bit more, if the default version gets defaulted into the api model someone would have to set `unstable` in order to use it which makes me think that this should just check specific versions.  The default case should never be hit unless there is programmatic error?  Maybe I'm thinking too hard on it :smile:

Reply:
I don't feel strong about doing any sort of error handling here since version errors are already be managed during validation.

Reply:
@jim-minter thoughts?

Reply:
I'm not worried about it.

Reply:
> I'm not worried about it.

Famous last words :smile: 

I'll just leave this existing example here (also noting that only specific versions are returned here, I'd argue they need another default handler and could/should make this something that can be passed back :smile: ).

```go
switch orchestratorType {
	case api.DCOS:
		switch orchestratorVersion {
		case api.DCOSVersion1Dot8Dot8:
			return dcosCustomData188
		case api.DCOSVersion1Dot9Dot0:
			return dcosCustomData190
		case api.DCOSVersion1Dot9Dot8:
			return dcosCustomData198
		case api.DCOSVersion1Dot10Dot0:
			return dcosCustomData110
		case api.DCOSVersion1Dot11Dot0:
			return dcos2CustomData111
		}
	default:
		// it is a bug to get here
		panic(fmt.Sprintf("BUG: invalid orchestrator %s", orchestratorType))
	}
	return ""
```

Reply:
OK, I don't mind switching to panic.

Reply:
updated